cypress/e2e/problem_creator.cy.ts

```diff
@@ -9,266 +9,290 @@ describe('Problem creator Test', () => {
     password: 'user',
   };
 
-  beforeEach(() => {
-    cy.clearCookies();
-    cy.clearLocalStorage();
+  // Define test modes
+  const testModes = [
+    {
+      name: 'Authenticated User',
+      isAuthenticated: true,
+      setup: () => {
+        cy.login(loginOptions);
+        profilePage.updatePreferredLanguage('es');
+      },
+      visit: () => {
+        cy.get('[data-nav-problems]').click();
+        cy.get('[data-nav-problems-create-options]').click();
+        cy.get('a[href="/problem/creator/"]').click();
+      },
+    },
+    {
+      name: 'Unauthenticated User',
+      isAuthenticated: false,
+      setup: () => {
+        // No setup needed for unauthenticated user
+      },
+      visit: () => {
+        cy.get('[data-nav-problems]').click();
+        cy.get('a[href="/problem/creator/"]').click();
+        cy.location('search').then((search) => {
+          if (!search.includes('lang')) {
+            cy.visit('/problem/creator/?lang=es');
+          }
```

Comment:
Same case as the previous comment. Here you can access the intermediate function and pass it the lang parameter:
```suggestion
        problemCreatorPage.visit('es');
```